### DISCUSSION TITLE
How to customize MRT_ShowHideColumnsMenu?

### DISCUSSION THREAD

**Discussion by fofyga** [KevinVandy/material-react-table#278] [Q&A]
What are best way to customize MRT_ShowHideColumnsMenu component? I try to change buttons and menu styles...

**Comment by KevinVandy** [comment]
Honestly, the columns menu is not very customizable. It is auto-generated depending on the features you have turned on or off.

What do you need to customize?

What you can do is totally replace the built in Show/Hide Columns Button and the menu with the `renderInternalToolbarButtons` prop

https://www.material-react-table.com/docs/guides/customize-toolbars#customize-built-in-internal-toolbar-button-area

**Reply by GabbereX** [reply]
> Honestly, the columns menu is not very customizable. It is auto-generated depending on the features you have turned on or off.
> 
> What do you need to customize?
> 
> What you can do is totally replace the built in Show/Hide Columns Button and the menu with the `renderInternalToolbarButtons`
> 
> https://www.material-react-table.com/docs/guides/customize-toolbars#customize-built-in-internal-toolbar-button-area

This solution is time-consuming. In my case, I only need to tweak the styles of this component slightly. It would be beneficial to have an option to modify the sx property for the MRT_ShowHideColumnsMenu wrapper. Otherwise, the component is well-implemented, but I've noticed that when the state changes, the container with the list decreases by 1 pixel for no apparent reason. In addition to this issue, I generally need to adjust the styles in this menu. Please provide this capability.
